components/Link.js

```diff
@@ -0,0 +1,26 @@
+import { setState } from "../index.js";
+import { routes } from "../routes.js";
+
+const Link = ({ href, label }) => {
+  let element;
+
+  const currentRoute = routes.find((route) => route.path === href);
+
+  if (currentRoute) {
+    if (currentRoute.isServerPage) {
+      element = document.createElement("a");
+      element.href = href;
+      element.textContent = label;
+    } else {
+      element = document.createElement("button");
+      element.textContent = label;
+      element.onclick = () => {
+        setState({ pathURLName: currentRoute.path });
+      };
+    }
```

Comment:
elementnya tetep pake link aja, tapi ketika diklik jalanin `event.preventDefault()` dulu 

```suggestion
      const element = document.createElement("a");
      element.href = href;
      element.textContent = label;
      element.onclick = (event) => {
        if (!currentRoute.isServerPage) {
          event.preventDefault();
          setState({ pathURLName: currentRoute.path });
        }
      }
      return element
```